File: programmers/study_1.py
def solution(answers):
    p_1 = [1,2,3,4,5]
    p_2 = [2,1,2,3,2,4,2,5]
    p_3 = [3,3,1,1,2,2,4,4,5,5]
    
    result = [0,0,0]
    count = 0
    
    for i in answers:
        if i == p_1[count%len(p_1)]:
            result[0] = result[0] + 1
        if i == p_2[count%len(p_2)]:
            result[1] = result[1] + 1
        if i == p_2[count%len(p_3)]:
            result[2] = result[2] + 1
        count += 1
        
    if min(result) == max(result):
        return [1,2,3]
    
    return result(index(max(result)))

File: programmers/study_1.scala
object Solution {
    def solution(answers: Vector[Int]): Vector[Int] = {
        
        val p1 = List(1,2,3,4,5)
        val p2 = List(2,1,2,3,2,4,2,5)
        val p3 = List(3,3,1,1,2,2,4,4,5,5)
        
        var result = List(0, 0, 0)
        
        for ((value, index) <- answers.zipWithIndex)
        {
            if (p1(index%p3.length) == value)
                result.patch(0, Seq(result(index%p1.length)+1), 1)
            if (p2(index%p3.length) == value)
                result.patch(1, Seq(result(index%p2.length)+1), 1)
            if (p3(index%p3.length) == value)
                result.patch(2, Seq(result(index%p2.length)+1), 1)
        }
        println (result)
        for ( i <- result)
        {
            print ("AAA")
            println (i)
        }
        return Vector[Int]()
    }
}